src/Tribe/Context.php

```diff
@@ -44,25 +44,29 @@ public function is_editing_post( $post_or_type = null ) {
 		}
 
 		if ( null !== $post_or_type ) {
+			$lookup = array( $_GET, $_POST, $_REQUEST );
+
+			$current_post = Tribe__Utils__Array::get_in_any( $lookup, 'post', get_post() );
+
 			if ( is_numeric( $post_or_type ) ) {
 
 				$post = $is_post ? get_post( $post_or_type ) : null;
 
-				return ! empty( $post ) && $post == get_post();
+				return ! empty( $post ) && $post == $current_post;
 			}
 
 			$post_types = is_array( $post_or_type ) ? $post_or_type : array( $post_or_type );
 
-			$post = $is_post ? get_post() : null;
+			$post = $is_post ? $current_post : null;
 
 			if ( count( array_filter( $post_types, 'is_numeric' ) ) === count( $post_types ) ) {
 				return ! empty( $post ) && in_array( $post->ID, $post_types );
 			}
 
-			if ( $is_post ) {
+			if ( $is_post && $post instanceof WP_Post ) {
 				$post_type = $post->post_type;
 			} else {
-				$post_type = Tribe__Utils__Array::get( $_GET, 'post_type', 'post' );
+				$post_type = Tribe__Utils__Array::get_in_any( $lookup, 'post_type', 'post' );
```

Comment:
Out of curiosity - any reason other than tidiness to not just use the `$lookup` array here since it's small?

Reply:
the idea is that we might want to change that in the future and changing it in one place only, on line 47, seems like the most maintainable solution